src/main/groovy/com/bmuschko/gradle/docker/DockerConventionJvmApplicationPlugin.groovy

```diff
@@ -132,23 +138,23 @@ abstract class DockerConventionJvmApplicationPlugin<EXT extends DockerConvention
         })
     }
 
-    private static Sync createSyncBuildContextTask(Project project, Dockerfile createDockerfileTask) {
-        project.tasks.create(SYNC_BUILD_CONTEXT_TASK_NAME, Sync, new Action<Sync>() {
+    private static TaskProvider<Sync> registerSyncBuildContextTask(Project project, TaskProvider<Dockerfile> createDockerfileTask) {
+        project.tasks.register(SYNC_BUILD_CONTEXT_TASK_NAME, Sync, new Action<Sync>() {
             @Override
             void execute(Sync sync) {
                 sync.with {
                     group = DockerRemoteApiPlugin.DEFAULT_TASK_GROUP
                     description = "Copies the distribution resources to a temporary directory for image creation."
                     dependsOn project.tasks.getByName(JavaPlugin.CLASSES_TASK_NAME)
-                    into(createDockerfileTask.destDir)
+                    into(createDockerfileTask.get().destDir)
```

Comment:
I don't think the `get()` is needed here.

Reply:
It is needed to get the `destDir` property which only exists on `Dockerfile` task and not `TaskProvider<Dockerfile>`

What happens when `dockerSyncBuildContext` is executed
1. `dockerSyncBuildContext` is configured
2. During configuration, the `createDockerfileTask.get()` call configures `dockerCreateDockerfile` task
3. `dockerCreateDockerfile` configuration is finished
4. `dockerSyncBuildContext` configuration is finished
5. `dockerSyncBuildContext` is executed
6. `dockerCreateDockerfile` is NOT executed. It only gets configured so we can calculate the `destDir` property correctly